### PR TITLE
account for logged in users on checkout

### PR DESCRIPTION
**Pull request by remstone7**

### CODE REVIEW

File: view/frontend/web/js/view/checkout/email.js [hunk 1]
```diff
@@ -1,40 +1,60 @@
 define([
     'uiComponent',
     'jquery',
-    'jquery/ui'
+    'jquery/ui',
+    '!domReady'
 ], function(Component) {
-    'use strict';
+    // initial
```
Comment (remstone7) [inline view/frontend/web/js/view/checkout/email.js]: will update this

File: view/frontend/web/js/view/checkout/email.js [hunk 2]
```diff
@@ -1,40 +1,61 @@
 define([
     'uiComponent',
     'jquery',
-    'jquery/ui'
+    'jquery/ui',
+    '!domReady'
```
Comment (klaviyojad) [inline view/frontend/web/js/view/checkout/email.js]: why is this needed?  I don't see anything in the added code pertaining to it

File: view/frontend/web/js/view/checkout/email.js [hunk 3]
```diff
@@ -1,40 +1,61 @@
...
 ], function(Component) {
     'use strict';
+    // initialize the customerData prior to returning the component
+    var _klaviyoCustomerData = window.customerData;
```
Comment (klaviyojad) [inline view/frontend/web/js/view/checkout/email.js]: what is window.customerData?

File: view/frontend/web/js/view/checkout/email.js (hunk 2 above)
Comment (remstone7) [reply view/frontend/web/js/view/checkout/email.js]: it makes the code not load until the document is ready. https://requirejs.org/docs/api.html#pageload

File: view/frontend/web/js/view/checkout/email.js (hunk 3 above)
Comment (remstone7) [reply view/frontend/web/js/view/checkout/email.js]: from our plugin, we put customerData on every page which contains a user's email if they're logged in

File: view/frontend/web/js/view/checkout/email.js [hunk 4]
```diff
@@ -1,40 +1,61 @@
...
+            });
+        },
+        postUserEmail: function (customer_email) {
+          var path = window.location.pathname;
+          if (path.slice(-1) == '/' ) {
+              path = path.slice(0, -1);
+          }
 
-                var url = window.location.protocol + '//' + window.location.host + path.substring(0,path.lastIndexOf("/"));
+          var url = window.location.protocol + '//' + window.location.host + path.substring(0,path.lastIndexOf("/"));
```
Comment (klaviyojad) [inline view/frontend/web/js/view/checkout/email.js]: why all the slicing and substring then appending, can we just do all this manipulation as part of one line instead of 3 lines? whatever it is is only used once in that ajax call as far as I can see

Comment (remstone7) [reply view/frontend/web/js/view/checkout/email.js]: looks like it's some cleaning, i don't know if i have a recommendation to do this in 1 line as it would get even more bulky - open to suggestions though!

File: view/frontend/templates/product/viewed.phtml [hunk 5]
```diff
@@ -4,8 +4,9 @@
 
 <script text="text/javascript">
   require(['domReady!'], function () {
-    _learnq.push(['track', 'Viewed Product', <?= $block->getViewedProductJson() ?>]);
-    _learnq.push(['trackViewedItem', <?= $block->getViewedItemJson() ?>]);
+    window._learnq = window._learnq || [];
```
Comment (remstone7) [inline view/frontend/templates/product/viewed.phtml]: adding the window here as i'd rather this not break if it has a loading problem

**Comment by ihor-sviziev** [comment]
@remstone7,
Would be great also send email to _learnq when guest customer fills it up on checkout. It will allow to associate customer flow on frontend (which products he/she saw) with info that later will be retrieved from API (quotes, orders)

**Comment by remstone7** [comment]
@ihor-sviziev will add that to next release!
